lib/streamlit/web/server/server.py

```diff
@@ -101,9 +102,24 @@ def start_listening(app: tornado.web.Application) -> None:
     port.  It will error after MAX_PORT_SEARCH_RETRIES attempts.
 
     """
+    ssl_options = None
```

Comment:
nit: Imagine we can't reassign values because `ssl_option` is constant. What would it take to make it happen? I guess, we would have to create a function:

```
def get_ssl_options(cert_file: SomeType, key_file: SomeType) -> Optional[SSLContext](https://docs.python.org/3/library/ssl.html#ssl.SSLContext)]:
```

That would have a few benefits:
* it would be easier to understand what type is `cert_file` and `key_file`. Now I don't know because `config.get_option` returns `Any`. That's why I wrote `SomeType` in a function signature but this could be replace with an actual type which I guess it either `str` or `File`.
* Would encapsulate this part (end error handling) better
* Would make code more readable.

Reply:
Done. Now we have `_get_ssl_options(cert_file: Optional[str], key_file: Optional[str]) -> Union[ssl.SSLContext, None]` function.